### PR TITLE
bring-back-feature(analytics): Added back the analytics visualizer, now in TS.

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Feature comeback**

This PR resolves [CX-1420]

### Description

Basically the same as the old, but now in TS. I also removed the stacktrack, it seems very much not needed.

New

https://user-images.githubusercontent.com/100233/125834701-566ac98b-84a6-4788-aded-d027da56931c.mov

Old for reference
IMG_4815

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Added analytics visualizer - pavlos, cx team

</details>

[CX-1420]: https://artsyproduct.atlassian.net/browse/CX-1420

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Added analytics visualizer - pavlos, cx team)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 13c86f2b9fc5541dc86339d5fde341caa6f6c3c3
</p>

File: src/palette/elements/Touchable/Touchable.tsx [hunk 1]
```diff
@@ -16,7 +14,6 @@ interface ExtraTouchableProps {
   flex?: number
   haptic?: HapticFeedbackTypes | true
   noFeedback?: boolean
-  useDefaultTouchable?: boolean
```
Comment (pvinis) [inline src/palette/elements/Touchable/Touchable.tsx:19]: side fix: this was causing a baaaad bug with layout of touchables. we should never use the `react-native-gesture-handler` touchables. they are bad.

File: src/lib/Components/Toast/Toast.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { GlobalStore } from "lib/store/GlobalStore"
+import { ToastOptions, ToastPlacement } from "./types"
+
+export const Toast = {
```
Comment (pvinis) [inline src/lib/Components/Toast/Toast.ts:4]: A way to access Toast globally from outside react components.

File: src/lib/store/ToastModel.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      id: state.sessionState.nextId,
+      message: newToast.message,
+      placement: newToast.placement,
+      ...newToast.options,
+    })
+    console.log(state.sessionState.toasts)
+  }),
+  _incrementNextId: action((state) => {
+    state.sessionState.nextId += 1
+  }),
```
Comment (pvinis) [inline src/lib/store/ToastModel.ts:63]: We found it ladies and gentlemen. A reason to hate easy-peasy 😅. The regular `add` means we have to update the state *and* increment. But actions only update state and thunks only call actions, so I named them with `_` to signify that we should not call these directly, we should use the regular `add` that will handle calling these.

File: src/palette/elements/Touchable/Touchable.tsx (hunk 1 above)
Comment (lordkiz) [reply src/palette/elements/Touchable/Touchable.tsx:19]: I agree. I never go near them.

**Review by lordkiz — APPROVED** [review]
Super! LGTM :shipit:

File: Artsy/View_Controllers/Admin/ARAdminSettingsViewController.m [hunk 4]
```diff
@@ -168,18 +167,6 @@ - (ARCellData *)showConsignmentsFlow
...
-    NSString *message = [AROptions boolForOption:AROptionsShowAnalyticsOnScreen] ? @"Stop" : @"Start";
-    NSString * title = NSStringWithFormat(@"%@ on Screen Analytics", message);
-
-    return [self tappableCellDataWithTitle:title selection:^{
-        BOOL current = [AROptions boolForOption:AROptionsShowAnalyticsOnScreen];
-        [AROptions setBool:!current forOption:AROptionsShowAnalyticsOnScreen];
-        exit(YES);
-    }];
-}
-
```
Comment (brainbicycle) [inline Artsy/View_Controllers/Admin/ARAdminSettingsViewController.m:182]: 🧹

**Review by brainbicycle — APPROVED** [review]

**Comment by brainbicycle** [comment]
🍞 !!!! Awesome

**Pull request merged**